### ISSUE TITLE
Dirt roads for motorcycle navigation.

### ISSUE THREAD

**Issue by molind** [valhalla/valhalla#4281]
I've received a few feature requests from our users and I'm looking at how difficult it could be. Motorcyclists often look for adventures and are ready to ride on tracks and some dirt paths. We already use `useTrails`, which adds priority to secondary roads, while they are asking about `highway=track` roads. What do you think about it? Does it make sense to add an extra option that unlocks track roads for navigation?

**Comment by nilsnolde** [comment]
Yeah motorcyclists often want some curious extras. IMO it’d make sense to allow preference of dirt roads (or rather avoidance of non-dirt roads?). 

Curvature is probably even more important? We have that available as edge property already IIRC and could be exposed. See https://kurviger.de for a Graphhopper based router.

**Comment by bmapsnavigation** [comment]
I use curvature in a local valhalla instance. Currently. and if I remember correctly, it builds with tests turned off or else the build fails. Still working on fixing that, but it's a slow progress as I find programming in C quite the challenge. My solution is an slightly adapted motorcyclecost.cc and a couple of other things I found here on the valhalla github when looking for how to add a custom cost, and besides testing during build, that actually works fine as far as I can see. There will always be a bit of discussion whether someone would prefer road A over B, because "A has more curves than B" though.

I would be interested to see if some kind of off road navigation could be build-in in valhalla.

**Comment by molind** [comment]
Here's the corrected version:

@nilsnolde Yes, I've seen it already. However, in this issue, I'd like to discuss the possibility of using `highway=track` for motorcycle routes. There are many tracks where driving motorcycles is legally permitted or not explicitly prohibited. We could probably use rules from https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access_restrictions. Tracks shouldn't be allowed by default, but if someone needs to go there and it's legal, we should be able to build a route.

**Comment by molind** [comment]
[Access restrictions](https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access_restrictions) mention Austria, Belarus, France and others. But that `use_tracks` definitely worth a try. Thanks for mentioning it!

**Comment by kevinkreiser** [comment]
looks related to this issue #4336 too

**Comment by nilsnolde** [comment]
I'll close here then, feel free to re-open if there's anything else

**Comment by molind** [comment]
Seems I was too happy to learn about that option. It worked for car, but doesn't work for motorcycle. Tried again today and it builds car route, but won't build route for motorcycle.

```
curl -X POST 'https://valhalla1.openstreetmap.de/route' \
  -H 'Content-Type: application/json' \
  -d '{
    "costing": "motorcycle",
    "costing_options": {
      "motorcycle": {
        "use_highways": 0.0,
        "use_trails": 1.0,
        "use_tracks": 1.0,
        "exclude_unpaved": false
      }
    },
    "locations": [
      {
        "lon": 27.3618185,
        "lat": 53.9179508
      },
      {
        "lon": 27.3722255,
        "lat": 53.9164469
      }
    ]
  }'
```

according to the docs. use_tracks is for `auto`, `bus`, `truck`. and `exclude_unpaved ` is for `auto`,`bus`,`taxi`, `truck`. No luck for motorcycle.

It works for `auto`:
```
curl -X POST 'https://valhalla1.openstreetmap.de/route' \
  -H 'Content-Type: application/json' \
  -d '{
    "costing": "auto",
    "locations": [
      {
        "lon": 27.3618185,
        "lat": 53.9179508
      },
      {
        "lon": 27.3722255,
        "lat": 53.9164469
      }
    ]
  }'
```

https://valhalla.openstreetmap.de/directions?profile=car&wps=27.3618185%2C53.9179508%2C27.3722255%2C53.9164469

**Comment by nilsnolde** [comment]
just wanted to say, why don't you? never realized you can only re-open when you were the one closing it.

don't rely on the docs, you'll need to go into the source. use_tracks is also used for motorbike costing. but clearly there's smth wrong.

**Comment by nvcnvn** [comment]
I have similar issue, here is my test case:
https://valhalla.openstreetmap.de/directions?profile=motorcycle&wps=106.80636763572694%2C10.81556536084666%2C106.80101931095123%2C10.813389203845343 

I need to make a small change so it work: https://github.com/nvcnvn/valhalla/commit/c6f82d55d3afed979bbab4b5f49b49c64b28f81c

**Comment by molind** [comment]
@nvcnvn Thanks. I'll check it.

**Comment by molind** [comment]
https://github.com/nvcnvn/valhalla/commit/c6f82d55d3afed979bbab4b5f49b49c64b28f81c with this change I was able to build route for a motorcycle.

**Comment by nilsnolde** [comment]
The change makes sense to me. Wanna PR that?

**Comment by molind** [comment]
Easy. It makes sense for me too.

**Comment by nilsnolde** [comment]
guess this is solved in #4348
